### DISCUSSION TITLE
How to use -parse_before= ?

### DISCUSSION THREAD

**Discussion by piersolenski** [uga-rosa/translate.nvim#34] [Q&A] [answered]
I'm unsure how `-parse_before=` works from the docs. Does it only take presets? For example, I would like strip a line of the characters `: ` if they exist, probably using a find and replace function. Thanks!

**Comment by uga-rosa** [comment, marked as answer]
There is a simple explanation in `:h translate-nvim-option-parse-before`, but was it difficult?
It can be called by registering it with setup as follows.
```lua
require("translate").setup({
  parse_before = {
    my_parser = {
      ---@param lines string[]
      ---@return string[]
      cmd = function(lines)
        for i = #lines, 1, -1 do
          if lines[i]:find(":") then
            table.remove(lines, i)
          end
        end
        return lines
      end,
    },
  },
})
```

```vim
:Translate EN -parse_before=my_parser
```

**Reply by piersolenski** [reply]
That's awesome, thanks for the clarification!
